src/script-child-process.ts

```diff
@@ -44,27 +45,33 @@ export type ScriptChildProcessState =
   | 'killing'
   | 'stopped';
 
-/**
- * A script config but the command is required.
- */
-type ScriptConfigWithRequiredCommand = ScriptConfig & {
-  command: Exclude<ScriptConfig['command'], undefined>;
-};
-
 /**
  * A child process spawned during execution of a script.
  */
 export class ScriptChildProcess {
-  private readonly _script: ScriptConfigWithRequiredCommand;
+  private readonly _script: ScriptReferenceWithCommand;
   private readonly _child: ChildProcessWithoutNullStreams;
+  private readonly _started = new Deferred<Result<void, SpawnError>>();
+  private readonly _completed = new Deferred<
+    Result<void, SpawnError | ExitSignal | ExitNonZero | Killed>
+  >();
   private _state: ScriptChildProcessState = 'starting';
 
+  /**
+   * Resolves when this process starts
+   */
+  get started(): Promise<Result<void, SpawnError>> {
+    return this._started.promise;
+  }
+
   /**
    * Resolves when this child process ends.
    */
-  readonly completed: Promise<
+  get completed(): Promise<
```

Comment:
can this just be:

```typescript
readonly completed = this._completed.promise;
```

Reply:
If there's even going to be a new run of the child process with new Deferreds for started and completion, then the getters are safer.

Reply:
Good point, done!

> If there's even going to be a new run of the child process with new Deferreds for started and completion, then the getters are safer.

The child process will only be re-used if it doesn't need to be restarted. Otherwise there will be a new `ScriptChildProcess` instance. So this is safe.